thetis/shallowwater_eq.py

```diff
@@ -370,10 +307,10 @@ def horizontal_viscosity(self, uv, nu, total_h):
 
             # Dirichlet bcs only for DG
             for bnd_marker in self.boundary_markers:
-                funcs = self.bnd_functions.get(bnd_marker)
+                funcs = bnd_conditions.get(bnd_marker)
                 ds_bnd = ds(int(bnd_marker))
                 if funcs is not None:
-                    eta_ext, uv_ext = self.get_bnd_functions(None, uv, bnd_marker)
+                    eta_ext, uv_ext = self.get_bnd_functions(eta, uv, bnd_marker, bnd_conditions)
```

Comment:
@stephankramer Note I had to change this: Some boundary conditions actually need elevation, and passing None will raise an error.